core/src/test/scala/org/apache/spark/rdd/DoubleRDDSuite.scala

```diff
@@ -18,10 +18,17 @@
 package org.apache.spark.rdd
 
 import org.scalatest.FunSuite
+import org.scalatest.Matchers._
 
 import org.apache.spark._
 
 class DoubleRDDSuite extends FunSuite with SharedSparkContext {
+  test("sum") {
+    sc.parallelize(Seq.empty[Double]).sum() should be(0.0 +- 0.0001)
```

Comment:
Doubles aren't unstable; `0.0 == 0.0` always. Yes i know what you mean but in these cases we can expect the result to exact to machine precision, always.